### PR TITLE
chore: clean up copy link post page onboarding v3 and social proof onboarding experiment

### PR DESCRIPTION
**Pull request by nensidosari**
Slack threads testifying these are ok to cleanup:

- [Copy link](https://dailydotdev.slack.com/archives/C06K3QEFPE1/p1709659131058209?thread_ts=1709630063.617649&cid=C06K3QEFPE1) (won)
- [post page onboarding v3](https://dailydotdev.slack.com/archives/C02AA18RD1D/p1710187214094689) (lost)
- [social proof onboarding](https://dailydotdev.slack.com/archives/C06K5KAHM0X/p1708948319189189?thread_ts=1708933869.441869&cid=C06K5KAHM0X) (won)

### CODE REVIEW

File: packages/shared/src/components/Feed.tsx [hunk 1]
```diff
@@ -293,35 +286,15 @@ export default function Feed<T>({
     }
   };
 
-  let lastShareMenuCloseTrackEvent = () => {};
+  const lastShareMenuCloseTrackEvent = () => {};
```
Comment (DragosIorgulescu) [inline packages/shared/src/components/Feed.tsx:289]: is this variable still being used?

**Review by DragosIorgulescu — APPROVED** [review]
just one non-blocking comment, otherwise, love seeing code removed ✂️

File: packages/shared/src/lib/featureManagement.ts [hunk 2]
```diff
@@ -41,8 +41,6 @@ const feature = {
     'post_page_onboarding',
```
Comment (rebelchris) [inline packages/shared/src/lib/featureManagement.ts:41]: Should this not also be removed then?

Comment (rebelchris) [reply packages/shared/src/lib/featureManagement.ts:41]: Oh has a follow up I guess?

**Review by rebelchris — APPROVED** [review]
Cool just note Dragos comment, seems valid :)

File: packages/shared/src/components/cards/ActionButtons.tsx [hunk 3]
```diff
@@ -51,12 +50,11 @@ type LastActionButtonProps = {
   onShare?: (post: Post) => unknown;
   onShareClick?: (event: React.MouseEvent, post: Post) => unknown;
   post: Post;
-  copyLinkFeature?: boolean;
 };
 function LastActionButton(props: LastActionButtonProps) {
-  const { onShareClick, post, copyLinkFeature } = props;
+  const { onShareClick, post } = props;
```
Comment (sshanzel) [inline packages/shared/src/components/cards/ActionButtons.tsx:55]: Should we be updating this function? I think it is a bit confusing to say `onShareClick` when what it does is copy the link.

File: packages/shared/src/components/cards/v1/ActionButtons.tsx [hunk 4]
```diff
@@ -20,19 +19,16 @@ import ConditionalWrapper from '../../ConditionalWrapper';
...
-import { useFeature } from '../../GrowthBookProvider';
-import { feature } from '../../../lib/featureManagement';
 
 interface ShareButtonProps {
   onShareClick?: (event: React.MouseEvent, post: Post) => unknown;
   post: Post;
-  copyLinkFeature?: boolean;
 }
 
 function ShareButton(props: ShareButtonProps) {
```
Comment (sshanzel) [inline packages/shared/src/components/cards/v1/ActionButtons.tsx:28]: Same here, I don't think we should be calling this ShareButton when what it does now is copy. Would confuse devs who has no background to this.

File: packages/shared/src/hooks/usePostContent.ts [hunk 5]
```diff
@@ -43,15 +40,8 @@ const usePostContent = ({
...
-    useSharePost(origin);
-  const onShare = () => {
-    if (copyLinkFeature) {
-      return copyLink(post);
-    }
-    return openSharePost(post);
-  };
-
+  const { sharePost, closeSharePost, copyLink } = useSharePost(origin);
+  const onShare = () => copyLink(post);
```
Comment (sshanzel) [inline packages/shared/src/hooks/usePostContent.ts:44]: Yep, this one too. We say `onShare` but we copy link 😅

**Review by sshanzel — CHANGES_REQUESTED** [review]
Looks great, just think we need more cleanup with the namings to avoid confusion.

**Comment by nensidosari** [comment]
@sshanzel agree worth changing the naming, am doing that ATM but will be quite a lot of changes. So thinking of raising a separate PR for that.

**Comment by nensidosari** [comment]
For the comments above @sshanzel created ticket, have started changes, have yet to finish them. In the meantime merging this PR.

**Review by sshanzel — APPROVED** [review]
Looks great to me! 🚀

**Pull request merged**